src/types/Slot0.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/*
+    Slot0 is a packed version of solidity structure:
+
+    // the current price
+    uint160 sqrtPriceX96;
+    // the current tick
+    int24 tick;
+    // protocol fee, expressed in hundredths of a bip
+    // upper 12 bits are for 1->0, and the lower 12 are for 0->1
+    // the maximum is 1000 - meaning the maximum protocol fee is 0.1%
+    // the protocolFee is taken from the input first, then the lpFee is taken from the remaining input
+    uint24 protocolFee;
+    // used for the lp fee, either static at initialize or dynamic via hook
+    uint24 lpFee;
+*/
```

Comment:
Could you maybe just also add a comment that we dont use a struct so that we get the benefits of a packed structure when we reference slot0 in memory/stack

Reply:
Added